core/src/Kokkos_Swap.hpp

```diff
@@ -40,18 +40,27 @@ namespace Impl {
 template <class T>
 struct is_swappable {
   template <class U>
-  static decltype(kokkos_swap(std::declval<T&>(), std::declval<T&>()))
+  KOKKOS_FUNCTION static decltype(kokkos_swap(std::declval<T&>(),
```

Comment:
Why did you add host device annotations?  What is this resolving and why is it part of this PR?

Reply:
Misreading of warning message which incidentally has warnings in Test_Array line 54, while both that file is changed and a change in line 54 of Kokkos_Swap.hpp occurred. I think the markup should be there, but this particular use of non-marked up function (decltype of function inside an expression to define a constexpr variable) is accepted by both clang and NVCC. However using decltype directly inside a device function is not:

https://godbolt.org/z/1vd36Tr9E

Reply:
So feel free to revert.